pkgs/by-name/su/surfer/package.nix

```
@@ -0,0 +1,65 @@
+{
+  lib,
+  fetchFromGitLab,
+  rustPlatform,
+  pkg-config,
+  openssl,
+  wayland,
+  autoPatchelfHook,
+  libxkbcommon,
+  libGL,
+  stdenv,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "surfer";
+  version = "0.2.0";
+
+  src = fetchFromGitLab {
+    owner = "surfer-project";
+    repo = "surfer";
+    rev = "v${version}";
+    hash = "sha256-C5jyWLs7fdEn2oW5BORZYazQwjXNxf8ketYFwlVkHpA";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    openssl
+    wayland
+    libxkbcommon
+    libGL
+    stdenv.cc.cc.libgcc
+  ];
```

```suggestion
    libX11
    libXcursor
    libXi
  ];
```

(and also the parameter list above)

These libraries are required for surfer to work on Xorg

---

Surfer also supports MacOS platform however I dont have the hardware to test it. Is there a tool or nix command where I can test the derivation for cross-platform/cross-architecture?

I think I need to use `stdenv.hostPlatform.emulator` or `binfmt` but I just couldn't find the documentation about using these in nixpkgs testing. Or maybe I should just spin up a Qemu VM?